### PR TITLE
[DOC] Tweaks for Array.new

### PR DESCRIPTION
**Pull request by BurdetteLamar**
- Rationalizes call-seq schemas.
- Reorganizes discussion into no-block/block.

### CODE REVIEW

File: array.c [hunk 1]
```diff
@@ -1060,48 +1060,41 @@ rb_ary_s_new(int argc, VALUE *argv, VALUE klass)
...
- *  the element for that +index+ is the return value from the block:
+ *  With a block given,
+ *  returns an +Array+ of the given +size+;
+ *  calls the block with each +index+ in the range <tt>(0..size-1)</tt>;
+ *  the element at that +index+ is the return value from the block:
  *
  *    a = Array.new(3) {|index| "Element #{index}" }
  *    a # => ["Element 0", "Element 1", "Element 2"]
  *
- *  Raises ArgumentError if +size+ is negative.
```
Comment (peterzhu2118) [inline array.c:1100]: Can we keep this one? I think we should describe what happens when `size` is negative.

Comment (BurdetteLamar) [reply array.c:1100]: Restored.

Comment (peterzhu2118) [reply array.c:1100]: Can we put this under the bullet point for `size`? I think it would make the most sense there.

Comment (BurdetteLamar) [reply array.c:1100]: Done.

**Comment by BurdetteLamar** [comment]
I've made some more changes here (hope not too burdensome to the review):
- Use outer bullets to emphasize the two cases:  no-block/block.
- Say 'array' instead of '+Array+ object', to keep it simple and casual (avoids the clutter of font-change).
- Move integer-convertible object remark to the end (applies to both no-block and block cases).
- Cite exceptions in the examples when feasible.
- Cite trivial case (size = 0) in examples.

Some of these ideas may belong as "considers" in the guidelines document?

**Review by peterzhu2118 — APPROVED** [review]

**Pull request merged**